### PR TITLE
Merge druntime Makefiles

### PR DESCRIPTION
**Pull request by kinke**
~~Depends on https://github.com/dlang/dmd/pull/15876.~~

### CODE REVIEW

**Comment by kinke** [comment]
The druntime **build** part is handled now by the generic `druntime/Makefile` (only small changes needed from original `druntime/posix.mak` to make it work on Windows too, see the first commit).

E.g., building compiler & druntime is a matter of invoking `make -j$(nproc)` in the repo root. On Windows too, with these prerequisites:
* Installed bash and common GNU tools, bundled with a Git for Windows installation.
* A GNU make, which AFAIK is NOT bundled with git. https://mirrors.ocf.berkeley.edu/gnu/make/make-4.2.1.tar.bz2 seems to be the latest version with an included Visual Studio `.sln` solution to easily build it from source. It's preinstalled on the Azure CI runners and the default `make` in PATH, as part of a mingw/msys installation in `C:\mingw64\bin`.
* For the druntime build to succeed, one currently needs to work in a matching MSVC environment (shell set up by `vcvarsall.bat`), as the C preprocessor otherwise fails to find system includes: https://issues.dlang.org/show_bug.cgi?id=24111

I had forgotten that DigitalMars make doesn't support parallelism via `-j`, so on top of getting rid of separate `win*.mak` Makefiles (supporting much less functionality than `posix.mak`) and the `copyimports.d` tool complication, the switch to GNU make also finally enables parallel druntime/Phobos builds & parallel test runs on Windows (probably mainly noticeable when compiling the Phobos unittests, once Phobos switched too).

---

The next step is merging the druntime **test** Makefiles.

**Comment by maxhaton** [comment]
Nice

File: compiler/src/dmd/target.d [hunk 1]
```diff
@@ -446,7 +446,7 @@ extern (C++) struct Target
             realsize = 10;
             realpad = 0;
             realalignsize = 2;
-            if (ptrsize == 4)
+            if (omfobj)
```
Comment (ibuclaw) [inline compiler/src/dmd/target.d:449]: This looks like a change for it's own pr.

File: compiler/test/tools/d_do_test.d [hunk 2]
```diff
@@ -618,9 +629,9 @@ string getDisabledReason(string[] disabledPlatforms, const ref EnvData envData)
 
 unittest
 {
-    immutable EnvData win32         = { os: "windows",  model: "32", };
-    immutable EnvData win32mscoff   = { os: "windows",  model: "32mscoff", };
-    immutable EnvData win64         = { os: "windows",  model: "64", };
+    immutable EnvData win32omf      = { os: "windows",  model: "32omf" };
+    immutable EnvData win32mscoff   = { os: "windows",  model: "32mscoff" };
+    immutable EnvData win64         = { os: "windows",  model: "64" };
```
Comment (ibuclaw) [inline compiler/test/tools/d_do_test.d:634]: Adding omf could also be factored out to its own pr as well.

File: druntime/test/aa/Makefile [hunk 3]
```diff
@@ -5,12 +5,12 @@ TESTS:=test_aa
 .PHONY: all clean
 all: $(addprefix $(ROOT)/,$(addsuffix .done,$(TESTS)))
 
-$(ROOT)/test_aa.done: $(ROOT)/%.done : $(ROOT)/%
+$(ROOT)/%.done: $(ROOT)/%$(DOTEXE)
```
Comment (ibuclaw) [inline druntime/test/aa/Makefile:8]: Adding of DOTEXE looks like it could be separated too.

File: compiler/src/dmd/target.d (hunk 1 above)
Comment (kinke) [reply compiler/src/dmd/target.d:449]: Well I would have had to disable some druntime tests temporarily on 32-bit Windows, because they test large structs and arrays. I don't remember which one(s).

**Comment by kinke** [comment]
The test Makefiles have been merged too, significantly increasing the test coverage on Windows. The Windows CI times don't benefit from the `-j` parallelism, because the tests are now run in both debug and release modes on Windows too (but that at least in parallel, so the CI timings didn't get worse either).

File: compiler/src/dmd/target.d (hunk 1 above)
Comment (thewilsonator) [reply compiler/src/dmd/target.d:449]: If you could split this off into a separate PR that would be much appreciated.

**Comment by thewilsonator** [comment]
LGTM

**Review by thewilsonator — APPROVED** [review]

**Comment by WalterBright** [comment]
This has caused a regression https://issues.dlang.org/show_bug.cgi?id=24347
